src/mlpack/methods/ann/layer/lstm.hpp

```diff
@@ -87,6 +87,19 @@ class LSTM
   template<typename InputType, typename OutputType>
   void Forward(InputType&& input, OutputType&& output);
 
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   * @param cellState Cell state of the LSTM.
+   * @param useCellState Use the cellState passed in the LSTM Cell.
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(InputType&& input, OutputType&& output,
+               OutputType&& cellState, bool useCellState);
```

Comment:
```c++
arma::mat in, out, cell;
lstm1.Forward(in, out, cell, false);
lstm2.Forward(in, out, cell, true);
lstm3.Forward(in, out, cell, false);
```

Here, lstm2 wants to use the cell state of lstm1, whereas lstm3 doesn't want to use lstm2's cell but cell is not empty. If I remove the bool then it will always rewrite the cell. (It would be not possible if remove the parameter all together). Thinking of which I added a param. May be there's another way.

Reply:
@zoq I'll like your views on this idea

Reply:
@zoq I'll incorporate other style checks